Sources/CacheAdvance/CacheReader.swift

```diff
@@ -52,6 +55,18 @@ final class CacheReader {
 
         switch try nextEncodedMessageSpan() {
         case let .span(messageLength):
+            // Check our assumptions before we try to read the message.
+            let endOfMessage = Bytes(startingOffset) + Bytes(MessageSpan.storageLength) + Bytes(messageLength)
```

Comment:
Can you help me understand why we need/want to cast to `Bytes`?

Reply:
`startingOffset` is `UInt64`
 `MessageSpan.storageLength` is `Int`
`messageLength` is `MessageSpan` (i.e. `UInt32`)

We need a common type, and `Bytes` (i.e. `UInt64`) seemed like the right move. That said, I can see an argument for utilizing `UInt64` directly. Let me do that.